### PR TITLE
Reset time to 0 on play after end instead of just mediaIndex

### PR DESCRIPTION
**Pull request by bc-bbay**
Replaces #294 (bad rebase)

### CODE REVIEW

File: src/videojs-hls.js [hunk 1]
```diff
@@ -950,6 +954,7 @@ videojs.Hls.prototype.drainBuffer = function(event) {
   // the playlist
   if (this.duration() !== Infinity && mediaIndex + 1 === playlist.segments.length) {
     this.mediaSource.endOfStream();
+    this.lastSeekedTime_ = undefined;
```
Comment (dmlap) [inline src/videojs-hls.js]: Why is this line necessary?

File: test/videojs-hls_test.js [hunk 2]
```diff
@@ -2254,6 +2260,8 @@ test('calling play() at the end of a video resets the media index', function() {
 
   player.play();
   strictEqual(player.hls.mediaIndex, 0, 'index is 1 after the first segment');
+
+  player.el().querySelector('.vjs-tech').vjs_getProperty = function() {};
```
Comment (dmlap) [inline test/videojs-hls_test.js]: This can be rewritten a bit more succinctly as `player.tech.el().vjs_getProperty`

File: src/videojs-hls.js [hunk 3]
```diff
@@ -318,7 +318,11 @@ videojs.Hls.prototype.setupMetadataCueTranslation_ = function() {
  */
 videojs.Hls.prototype.play = function() {
   if (this.ended()) {
-    this.mediaIndex = 0;
+    // If this.lastSeekedTime_ is defined, a seek has happened after playback ended, as it is set undefined at video end.
+    // We should begin playback from that point.
```
Comment (dmlap) [inline src/videojs-hls.js]: The code below makes sense but this comment confuses me.

File: src/videojs-hls.js (hunk 1 above)
Comment (bc-bbay) [inline src/videojs-hls.js]: So that the condition above returns true, and we know to reset. It's not wiped out automatically at the end of playback, but continues to store the last seeked value (if the user ever seeked at all).

File: src/videojs-hls.js (hunk 3 above)
Comment (bc-bbay) [inline src/videojs-hls.js]: I'll reword it.

File: test/videojs-hls_test.js (hunk 2 above)
Comment (bc-bbay) [inline test/videojs-hls_test.js]: Will update

File: src/videojs-hls.js (hunk 1 above)
Comment (bc-bbay) [inline src/videojs-hls.js]: Sorry, you're right, that's both the correct behavior and what is actually happening. But null != undefined and still affects the reset at post-end playback.

Comment (dmlap) [inline src/videojs-hls.js]: In that case, I think we should be consistent about what value we choose for `lastSeekedTime_` to indicate "unset". What do you think about initializing it to `null` in the init function and then just checking for null?

Comment (bc-bbay) [inline src/videojs-hls.js]: Would we then not unset to null after a seek completes? Because the question I'm asking is, "has there been a seek since the video ended", and if video ending and seek completing do the same thing, I can't get the answer.

Comment (dmlap) [inline src/videojs-hls.js]: If you seek once the video has ended, `ended()` should be false.
